src/utils/transforms.cljs

```diff
@@ -21,3 +24,37 @@
   [json]
   (when-not (= json "undefined")
     (try (.parse js/JSON json) (catch js/Error _ (when (string? json) json)))))
+
+;; Functions took from reagent and modified to also transform vectors contained in styles
+
+(def ^:private js-val? reagent.impl.util/js-val?)
+(def ^:private named? reagent.impl.util/named?)
+(def ^:private cached-prop-name reagent.impl.template/cached-prop-name)
+(declare styles-with-vectors)
+
+(defn ^:private kv-conv
+  [o k v]
+  (doto o
+    (oops/gobj-set (cached-prop-name k) (styles-with-vectors v))))
+
+(defn styles-with-vectors
+  [x]
+  (cond (js-val? x) x
+        (named? x)  (name x)
+        (map? x)    (reduce-kv kv-conv #js {} x)
+        (vector? x) (to-array (map styles-with-vectors x))
+        (coll? x)   (clj->js x)
+        (ifn? x)    (fn [& args]
+                      (apply x args))
+        :else       (clj->js x)))
+
+(defn copy-js-obj-to-map
+  "Copy `obj` keys and values into `m` if `(pred obj-key)` is satisfied."
+  [obj m pred]
+  (persistent!
+   (reduce (fn [acc js-prop]
+             (if (pred js-prop)
+               (assoc! acc js-prop (oops/gobj-get obj js-prop))
+               acc))
+           (transient m)
+           (js-keys obj))))
```

Comment:
Used persistent! and transient because it was really easy to add it,... and this is a general purpose function.

Reply:
Have you seen these benchmarks related to `specter`? There's one particular benchmark showing that transients on small(ish) maps can be faster with just `reduce-kv`. In that particular benchmark, `specter` performs significantly better than with transients or the vanilla immutable processing.

So moral of the story is, transients may not be always faster. How small can a map be until the transients pay off in our mobile stack? I don't know.

https://github.com/redplanetlabs/specter/blob/master/scripts/benchmarks.clj#L118

https://gist.github.com/nathanmarz/b7c612b417647db80b9eaab618ff8d83#file-results-clojure-1-8-0-clj-L34-L37

Reply:
Thanks again @ilmotta because of the reasoining on this!

My measurings weren't as rigorous as them, but tested it, and this approach was giving smaller timings than not using it, the difference wasn't too much, but existed.
Maybe in some cases it will take longer than just not using it :thinking: 

A good question for the performance crew I think.